src/components/feedback/feedbackModal.tsx

```diff
@@ -0,0 +1,235 @@
+  width: 500px;
+  max-width: 100%;
+  max-height: calc(100% - 64px);
+  display: flex;
+  flex-direction: column;
+  box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.05), 0 4px 16px rgba(0, 0, 0, 0.2);
+  transition: transform 0.2s ease-in-out;
+  transform: translate(0, 0) scale(1);
+  dialog:not([open]) & {
+    transform: translate(0, -16px) scale(0.98);
+  }
+`;
+
+const Header = styled.h2`
+  font-size: 20px;
+  font-weight: 600;
+  border-bottom: 1px solid #ccc;
+  padding: 20px 24px;
+  margin: 0px;
+`;
+
+const Form = styled.form`
+  display: flex;
+  overflow: auto;
+  flex-direction: column;
+  gap: 16px;
+  padding: 24px;
+`;
+
+const Label = styled.label`
+  display: flex;
+  flex-direction: column;
+  gap: 4px;
+  margin: 0px;
+`;
+
+const inputStyles = css`
+  border: 1px solid #ccc;
+  border-radius: 4px;
+  font-size: 14px;
+  padding: 6px 8px;
+  &:focus {
+    outline: 1px solid rgba(108, 95, 199, 1);
+    border-color: rgba(108, 95, 199, 1);
+  }
+`;
+
+const Input = styled.input`
+  ${inputStyles}
+`;
+
+const TextArea = styled.textarea`
+  ${inputStyles}
+  min-height: 64px;
+  resize: vertical;
+`;
+
+const ModalFooter = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  gap: 8px;
+  margin-top: 8px;
+`;
+
+const buttonStyles = css`
+  border: 1px solid #ccc;
+  border-radius: 4px;
+  cursor: pointer;
+  font-size: 14px;
+  font-weight: 600;
+  padding: 6px 16px;
+`;
+
+const SubmitButton = styled.button`
+  ${buttonStyles}
+  background-color: rgba(108, 95, 199, 1);
+  color: #fff;
+  &:hover {
+    background-color: rgba(88, 74, 192, 1);
+  }
+  &:focus-visible {
+    outline: 1px solid rgba(108, 95, 199, 1);
+    background-color: rgba(88, 74, 192, 1);
+  }
+`;
+
+const CancelButton = styled.button`
+  ${buttonStyles}
+  background-color: #fff;
+  color: #231c3d;
+  font-weight: 500;
+  &:hover {
+    background-color: #eee;
+  }
+  &:focus-visible {
+    outline: 1px solid rgba(108, 95, 199, 1);
+    background-color: #eee;
+  }
+`;
+
+const FlexColumns = styled.div`
+  display: flex;
+  flex-direction: row;
+  gap: 16px;
+  & > * {
+    flex: 1;
+  }
+`;
+
+interface FeedbackModalProps {
+  onClose: () => void;
+  onSubmit: (data: {comment: string; email: string; name: string}) => void;
+  open: boolean;
+}
+
+function stopPropagation(e: React.MouseEvent) {
+  e.stopPropagation();
+}
+
+const retrieveStringValue = (formData: FormData, key: string) => {
+  const value = formData.get(key);
+  if (typeof value === 'string') {
+    return value.trim();
+  }
+  return '';
+};
+
+export function FeedbackModal({open, onClose, onSubmit}: FeedbackModalProps) {
+  const dialogRef = React.useRef<HTMLDialogElement>(null);
+  const formRef = React.useRef<HTMLFormElement>(null);
+
+  useFocusTrap(dialogRef, open);
+  useShortcut('Escape', onClose);
+
+  // Reset on close
+  useEffect(() => {
+    if (!open) {
+      setTimeout(() => {
+        formRef.current.reset();
+      }, 200);
+    }
+  }, [open]);
```

Comment:
nit: should we clear the timer by returning a function from useEffect?

Reply:
done